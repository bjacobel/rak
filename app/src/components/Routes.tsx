import { Switch, Route } from 'wouter';

import Main from './Main';
import Child from './Child';
import Error from './Error';
import NotFound from './NotFound';

export default () => (
  <Switch>
    <Route component={Main} path="/" />
    <Route component={Child} path="/child/:id" />
    <Route component={Error} path="/error" />
    <Route component={NotFound} path="*" />
  </Switch>
);
