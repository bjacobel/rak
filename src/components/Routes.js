import React, { Component } from 'react';
import {
  BrowserRouter,
  Route,
  Switch,
} from 'react-router-dom';

import NotFound from './NotFound';
import Child from './Child';
import Main from './Main';

export default class Routes extends Component {
  render() {
    return (
      <BrowserRouter>
        <div>
          <Switch>
            <Route path="/" exact component={ Main } />
            <Route path="/child/:id" component={ Child } />
            <Route component={ NotFound } />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}
