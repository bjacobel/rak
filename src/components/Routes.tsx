import React from 'react';
import { Routes, Route } from 'react-router-dom';

import NotFound from './NotFound';
import Child from './Child';
import Main from './Main';

export default () => (
  <Routes>
    <Route element={<Main />} path="/" />
    <Route element={<Child />} path="/child/:id" />
    <Route element={<NotFound />} path="*" />
  </Routes>
);
