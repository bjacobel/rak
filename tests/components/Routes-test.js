import React from 'react';
import { shallow, mount } from 'enzyme';

import Routes from '../../src/components/Routes';

jest.mock('../../src/components/Main');
jest.mock('../../src/components/Child');
jest.mock('../../src/components/NotFound');

const setPath = (value) => {
  // thanks @cpojer! https://github.com/facebook/jest/issues/890#issuecomment-209698782
  Object.defineProperty(window.location, 'pathname', {
    writable: true,
    value,
  });
};

describe('Routes component', () => {
  it('matches snapshot', () => {
    expect(shallow(<Routes />)).toMatchSnapshot();
  });

  it('has a home route', () => {
    setPath('/');
    const routes = mount(<Routes />);
    expect(routes.find('Connect(Main)').length).toBe(1);
  });

  it('has a route for child views that takes a url param', () => {
    setPath('/child/1');
    const routes = mount(<Routes />);
    expect(routes.find('Child').length).toBe(1);
    expect(routes.find('Child').props()).toEqual(
      expect.objectContaining({
        match: expect.objectContaining({
          params: {
            id: '1',
          },
        }),
      }),
    );
  });

  it('has a fallthrough 404', () => {
    setPath('/asdfasaddf');
    const routes = mount(<Routes />);
    expect(routes.find('NotFound').length).toBe(1);
  });
});
