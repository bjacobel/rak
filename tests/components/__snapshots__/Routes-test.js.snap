// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component matches snapshot 1`] = `
<BrowserRouter>
  <div>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/child/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
